include/deal.II/lac/petsc_matrix_base.h

```diff
@@ -784,6 +784,44 @@ namespace PETScWrappers
     void Tvmult_add (VectorBase       &dst,
                      const VectorBase &src) const;
 
+    /**
+     * Perform the matrix-matrix multiplication $C = AB$, or, if an
```

Comment:
an -> a

Reply:
As I said, if we move it into the sparse matrix class we will have to implement it separately for `SparseMatrix` and `MPI::SparseMatrix`. Do you still think it is better than putting it in `MatrixBase`?

Reply:
What if you put the function you implemented into the `private:` section of the base class, and then derived classes would have similar functions that simply forward to the private function of the base class? This way, you would still have control over the concrete types in the `public` interface, but you wouldn't have to duplicate the actual implementation.

If you agree that this is the way to go, then we can merge this patch as a first step, and leave making the function protected/calling it from derived classes for a second step.

Reply:
I meant `protected:` instead of `private:`, of course.

Reply:
It's done.